### PR TITLE
[#133] Add option to core builder to enable ASAN

### PR DESCRIPTION
**Pull request by alanking**
Addresses #133

The CMake option defaults to `OFF`, but I just went with `YES`/`NO` to match the other options. `ON`/`YES` and `OFF`/`NO` are equivalent, as I understand it.

### CODE REVIEW

File: build_and_copy_packages_to_dir.sh [hunk 1]
```diff
@@ -84,6 +86,7 @@ build_jobs=0
 debug_config="-DCMAKE_BUILD_TYPE=Release"
 unit_test_config="-DIRODS_UNIT_TESTS_BUILD=YES"
 msi_test_config="-DIRODS_MICROSERVICE_TEST_PLUGINS_BUILD=YES"
+enable_asan="-DIRODS_MICROSERVICE_TEST_PLUGINS_BUILD=NO"
```
Comment (alanking) [inline build_and_copy_packages_to_dir.sh:89]: Whoooooops :)

Comment (alanking) [reply build_and_copy_packages_to_dir.sh]: Done

**Review by korydraughn — COMMENTED** [review]
Excellent. Does it work?

**Comment by alanking** [comment]
Turning it on works, yep. The default (that is, leaving it off) was of course messed up before the 2nd commit, so I haven't tried that yet.

**Comment by korydraughn** [comment]
Got it.

**Comment by alanking** [comment]
I just confirmed that the option turns the option on when supplied and leaves ASAN off when not supplied, so I went ahead and squashed.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Pull request merged**